### PR TITLE
feat: Add maskable option field for profile property settings - EXO-68604 - Meeds-io/MIPs#111

### PR DESCRIPTION
**Pull request by hakermi**
Add non-maskable option field for profile property settings

### CODE REVIEW

File: component/core/src/main/java/org/exoplatform/social/core/profileproperty/ProfilePropertyServiceImpl.java [hunk 1]
```diff
@@ -42,11 +42,14 @@ public class ProfilePropertyServiceImpl implements ProfilePropertyService, Start
...
 
-  private static final String SYNCHRONIZED_DISABLED_PROPERTIES= "synchronizationDisabledProperties";
+  private static final String                        SYNCHRONIZED_DISABLED_PROPERTIES    = "synchronizationDisabledProperties";
 
   protected List<ProfilePropertyDatabaseInitializer> profielPropertyPlugins              = new ArrayList<>();
 
-  private  List<String>                         synchronizedGroupDisabledProperties = new ArrayList<>();
+  private List<String>                               synchronizedGroupDisabledProperties = new ArrayList<>();
+
+  private static final List<String>                  NON_HIDDENABLE_PROPS                =
```
Comment (ahamdi) [inline component/core/src/main/java/org/exoplatform/social/core/profileproperty/ProfilePropertyServiceImpl.java]: We may nee to make those as init params to be able to add others by configuration if needed.

File: component/core/src/main/resources/db/changelog/social-rdbms.db.changelog-1.0.0.xml [hunk 2]
```diff
@@ -914,4 +914,11 @@
         </sql>
     </changeSet>
 
+    <changeSet author="social" id="1.0.0-93">
+        <addColumn tableName="SOC_PROFILE_PROPERTY_SETTING">
+            <column name="IS_HIDDENABLE" type="BOOLEAN" defaultValue="false">
```
Comment (ahamdi) [inline component/core/src/main/resources/db/changelog/social-rdbms.db.changelog-1.0.0.xml]: shouldn't this be true as default value ?

File: webapp/portlet/src/main/resources/locale/portlet/Portlets_en.properties [hunk 3]
```diff
@@ -698,6 +698,9 @@ profileSettings.label.multiValued=Multi valued
 profileSettings.label.groupSynchronized=Group synchronized
 profileSettings.label.required=Required
 profileSettings.label.active=Active
+profileSettings.label.maskable=Maskable
```
Comment (ahamdi) [inline webapp/portlet/src/main/resources/locale/portlet/Portlets_en.properties]: I am not sure which will be a better wording ? hiddenable or maskable ?

File: component/core/src/main/java/org/exoplatform/social/core/profileproperty/ProfilePropertyServiceImpl.java (hunk 1 above)
Comment (hakermi) [reply component/core/src/main/java/org/exoplatform/social/core/profileproperty/ProfilePropertyServiceImpl.java]: OK

File: component/core/src/main/resources/db/changelog/social-rdbms.db.changelog-1.0.0.xml (hunk 2 above)
Comment (hakermi) [reply component/core/src/main/resources/db/changelog/social-rdbms.db.changelog-1.0.0.xml]: I don't know,
it is a functional question

File: webapp/portlet/src/main/resources/locale/portlet/Portlets_en.properties (hunk 3 above)
Comment (hakermi) [reply webapp/portlet/src/main/resources/locale/portlet/Portlets_en.properties]: a functional question, in the task they used the label Maskable

File: component/core/src/main/java/org/exoplatform/social/core/profileproperty/ProfilePropertyServiceImpl.java (hunk 1 above)
Comment (hakermi) [reply component/core/src/main/java/org/exoplatform/social/core/profileproperty/ProfilePropertyServiceImpl.java]: Done

**Review by ahamdi — APPROVED** [review]

**Pull request merged**